vendor/engines/saas/lib/saas/engine.rb

```diff
@@ -17,6 +17,20 @@ class Engine < ::Rails::Engine
         })
       end
 
+      LoginController.class_eval do
+        before_action :page_job, only: [:public, :private]
+
+        :private
+
+        def page_job
+          auth_request = request.params['code'] && request.params['state']
+          if request.referer !~ /github\.com/ && !logged_in? && !auth_request
```

Comment:
:OK: let's leave it. This is in the SaaS engine so Enterprise should not be affected, correct?